ls1intum/Artemis
# User-statistics-fix

### Checklist

### Motivation and Context


The Month and year displayed in the Month and Year Tab was chosen by the Month/Year of the very right date of the chart. Since this can be confusing, this should be changed.

### Description

- The Year displayed in the year tab is now the year of the 7th bar, which is the right side of the middle of the chart
- The Month displayed in the Month tab is now depending on the length of the month. Its always the right bar next to the middle of the chart:
       If the length of the month is odd, like December, the month of the 17th bar is chosen
       If the length of month is even, like November, the month of the 16th. bar is chosen as the right center of the chart

(for months with odd lengths, choosing the middle bar would lead to never displaying 30 day months like November, August etc., hence the bar right next to the middle is chosen)

### Steps for Testing


1. Log in to Artemis
2. Navigate to Server Administration -> User statistics
3. Make sure that the above mentioned months and years are shown

(edge cases are hard to test, you could check out the branch locally and play around with the time of your pc, like seen in the screenshot)

### Screenshots


November has 30 days:
<img width="1402" alt="Screenshot 2021-01-15 at 14 46 24" src="https://user-images.githubusercontent.com/44401560/103653993-98df6300-4f65-11eb-9086-a390681762d0.png">


December has 31 days:
When it's the 14.01.2021:
<img width="1403" alt="Screenshot 2021-01-14 at 14 49 45" src="https://user-images.githubusercontent.com/44401560/103654060-b3b1d780-4f65-11eb-90ed-32403341afac.png">
When it's the 15.01.2021:
<img width="1402" alt="Screenshot 2021-01-15 at 14 48 22" src="https://user-images.githubusercontent.com/44401560/103654091-bca2a900-4f65-11eb-8325-9a700bb17b00.png">


Year:
<img width="1402" alt="Screenshot 2021-01-14 at 14 50 34" src="https://user-images.githubusercontent.com/44401560/103654122-c5937a80-4f65-11eb-9247-4b75c86d4dd8.png">
